R-package/R/aliases.R

```diff
@@ -7,119 +7,46 @@
-            , "label_column" = c(
-                "label_column"
-                , "label"
-            )
-            , "linear_tree" = c(
-                "linear_tree"
-                , "linear_trees"
-            )
-            , "max_bin" = c(
-                "max_bin"
-                , "max_bins"
-            )
-            , "max_bin_by_feature" = "max_bin_by_feature"
-            , "min_data_in_bin" = "min_data_in_bin"
-            , "pre_partition" = c(
-                "pre_partition"
-                , "is_pre_partition"
-            )
-            , "precise_float_parser" = "precise_float_parser"
-            , "two_round" = c(
-                "two_round"
-                , "two_round_loading"
-                , "use_two_round_loading"
-            )
-            , "use_missing" = "use_missing"
-            , "weight_column" = c(
-                "weight_column"
-                , "weight"
-            )
-            , "zero_as_missing" = "zero_as_missing"
-        )
-    )
+    all_aliases <- .PARAMETER_ALIASES()
+    return(all_aliases[c(
+        "bin_construct_sample_cnt"
+        , "categorical_feature"
+        , "data_random_seed"
+        , "enable_bundle"
+        , "feature_pre_filter"
+        , "forcedbins_filename"
+        , "group_column"
+        , "header"
+        , "ignore_column"
+        , "is_enable_sparse"
+        , "label_column"
+        , "linear_tree"
+        , "max_bin"
+        , "max_bin_by_feature"
+        , "min_data_in_bin"
+        , "pre_partition"
+        , "precise_float_parser"
+        , "two_round"
+        , "use_missing"
+        , "weight_column"
+        , "zero_as_missing"
+    )])
 }
 
 # [description] List of respected parameter aliases. Wrapped in a function to take advantage of
 #               lazy evaluation (so it doesn't matter what order R sources files during installation).
 # [return] A named list, where each key is a main LightGBM parameter and each value is a character
 #          vector of corresponding aliases.
 .PARAMETER_ALIASES <- function() {
-    learning_params <- list(
-        "boosting" = c(
-            "boosting"
-            , "boost"
-            , "boosting_type"
-        )
-        , "early_stopping_round" = c(
-            "early_stopping_round"
-            , "early_stopping_rounds"
-            , "early_stopping"
-            , "n_iter_no_change"
-        )
-        , "num_iterations" = c(
-            "num_iterations"
-            , "num_iteration"
-            , "n_iter"
-            , "num_tree"
-            , "num_trees"
-            , "num_round"
-            , "num_rounds"
-            , "nrounds"
-            , "num_boost_round"
-            , "n_estimators"
-            , "max_iter"
+    aliases <- jsonlite::fromJSON(
+        .Call(
+            LGBM_DumpParamAliases_R
         )
     )
-    return(c(learning_params, .DATASET_PARAMETERS()))
+    for (alias in names(aliases)) {
+        aliases[[alias]] <- c(aliases[[alias]], alias)
+    }
```

Comment:
@jameslamb Could you please help with R code here? The failing test is
```
  == Failed ======================================================================
  -- 1. Failure (test_parameters.R:60:3): .PARAMETER_ALIASES() returns a named lis
  all(sapply(param_aliases, is.character)) is not TRUE
  
  `actual`:   FALSE
  `expected`: TRUE 
```
and the problem is in keys with empty lists:
```R
aliases <- jsonlite::fromJSON("{\"task\": [\"task_type\", \"test\"], \"config\": [\"config_file\"], \"boosting\": [\"boosting_type\", \"boost\"], \"pre_filter\": []}")

for (alias in names(aliases)) {
    aliases[[alias]] <- c(aliases[[alias]], alias)
}

all(sapply(aliases, is.character))
length(unique(names(aliases))) == length(aliases)
print(aliases)
```
```
[1] FALSE
[1] TRUE
$task
[1] "task_type" "test"      "task"     

$config
[1] "config_file" "config"     

$boosting
[1] "boosting_type" "boost"         "boosting"     

$pre_filter
$pre_filter[[1]]
[1] "pre_filter"
```
Note `pre_filter` in the example above.

Reply:
Looks like I've managed to fix this, but will be happy to see a more elegant way 🙂